.github/workflows/ci-app-infra-service.yml

```
@@ -16,11 +16,26 @@ on:
       - infra/test/**
       - .github/workflows/ci-app-infra-service.yml
   workflow_dispatch:
+    inputs:
+      version:
+        required: true
+        default: "main"
+        description: Tag or branch or SHA to test
 
 jobs:
+  # `make infra-test-service` will build and publish the release itself, but
+  #  doing it explicitly with the re-usable workflow helps avoid race conditions
+  build-and-publish:
+    name: Build
+    uses: ./.github/workflows/build-and-publish.yml
+    with:
+      app_name: app
+      ref: ${{ inputs.version || github.event.pull_request.head.sha || 'main' }}
```

nit: `github.event.pull_request.head.sha` only works for the `pull_request` trigger and won't work for the `push` trigger. It's coincidence that the hardcoded `'main'` ref works for the `push` trigger since the `push` trigger coincidentally is configured to only trigger with `branches: [main]`, but that's not as DRY as it could be. Consider adding `github.ref` to this chain, potentially just replacing `github.event.pull_request.head.sha`.

```suggestion
      ref: ${{ inputs.version || github.ref || 'main' }}
```

or 

```
      ref: ${{ inputs.version || github.event.pull_request.head.sha || github.ref || 'main' }}
```